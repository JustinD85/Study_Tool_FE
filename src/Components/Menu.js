import React, { Component } from 'react';

class Menu extends Component {

  checkClassName(elementName) {
    const { currentView } = this.props;
    return currentView === elementName ? 'active-view': undefined;
  }

  render() {
    const { handleContentView, handleAppView, user } = this.props;

 
    return (
      <nav className="Menu">
        <h1>Js Fishing</h1>
        <ul>
          <li
            onClick={(e) => handleContentView('profile', e)}
            className={this.checkClassName('profile')}
          >
            {user.firstName + '\'s Profile'}
          </li>
          <li
            onClick={(e) => handleContentView('train', e)}
            className={this.checkClassName('train')}
          >
            Train
        </li>
          <li onClick={(e) => handleContentView('game', e)}
          className={this.checkClassName('game')}
          >
            Play
        </li>
        </ul>
        <button onClick={(e) => handleAppView('login',e)}>Back</button>
      </nav>
    )
  }
}

export default Menu;