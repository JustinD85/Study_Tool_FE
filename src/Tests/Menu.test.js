import React from 'react';
import Menu from '../Components/Menu';
import { shallow } from 'enzyme';
import user from './mockUser'

describe('<Menu/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Menu user={user}/>
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});