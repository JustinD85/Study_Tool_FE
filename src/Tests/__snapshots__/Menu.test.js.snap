// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu/> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    user={
      Object {
        "user": Object {
          "cascading_style_sheets": Object {
            "words": Array [],
          },
          "email": "me@me.com",
          "firstName": "Justin",
          "html_elements": Object {
            "words": Array [],
          },
          "javascript": Object {
            "words": Array [
              "var",
            ],
          },
          "lastName": "Duncan",
          "mod_one_vocab": Object {
            "words": Array [],
          },
          "score": 150,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Js Fishing
        </h1>,
        <ul>
          <li
            onClick={[Function]}
          >
            undefined's Profile
          </li>
          <li
            onClick={[Function]}
          >
            Train
          </li>
          <li
            onClick={[Function]}
          >
            Play
          </li>
        </ul>,
        <button
          onClick={[Function]}
        >
          Back
        </button>,
      ],
      "className": "Menu",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Js Fishing",
        },
        "ref": null,
        "rendered": "Js Fishing",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              onClick={[Function]}
            >
              undefined's Profile
            </li>,
            <li
              onClick={[Function]}
            >
              Train
            </li>,
            <li
              onClick={[Function]}
            >
              Play
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "undefined's Profile",
              "className": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "undefined's Profile",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Train",
              "className": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Train",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Play",
              "className": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Play",
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Back",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Back",
        "type": "button",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Js Fishing
          </h1>,
          <ul>
            <li
              onClick={[Function]}
            >
              undefined's Profile
            </li>
            <li
              onClick={[Function]}
            >
              Train
            </li>
            <li
              onClick={[Function]}
            >
              Play
            </li>
          </ul>,
          <button
            onClick={[Function]}
          >
            Back
          </button>,
        ],
        "className": "Menu",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Js Fishing",
          },
          "ref": null,
          "rendered": "Js Fishing",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                onClick={[Function]}
              >
                undefined's Profile
              </li>,
              <li
                onClick={[Function]}
              >
                Train
              </li>,
              <li
                onClick={[Function]}
              >
                Play
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "undefined's Profile",
                "className": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "undefined's Profile",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Train",
                "className": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Train",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Play",
                "className": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Play",
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Back",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Back",
          "type": "button",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
